[Zhang] iP

gnahzeus opened the pull request:
### Zeus

> “A goal without a plan is just a wish.” ― Antoine de Saint-Exupéry ([source](https://www.goodreads.com/quotes/87476-a-goal-without-a-plan-is-just-a-wish))

Zeus turns your wishes into plans by managing your everyday tasks. It's

- text-based
- easy to pick up
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🎉 

And it is **FREE**!

Features: 

- [x] Managing tasks
- [ ] GUI (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```Java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

gnahzeus committed:
implement greet, echo, exit

gnahzeus committed:
implement add, list

gnahzeus committed:
implement Level 3. Mark as Done

gnahzeus committed:
implement Level 4. ToDos, Events, Deadlines

gnahzeus committed:
automate text UI testing

gnahzeus committed:
implement Level 5. Handle Errors

gnahzeus committed:
implement Level 6.Delete

gnahzeus committed:
add javadoc comments to public methods/classes

gnahzeus committed:
Implement Level 7. Save

gnahzeus committed:
Implement Level 8. Dates and Times

gnahzeus committed:
Fix level 7 unwanted changes

gnahzeus committed:
Remove extra import

gnahzeus committed:
Merge branch 'branch-Level-7'

gnahzeus committed:
Merge branch 'branch-Level-8'

gnahzeus committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

gnahzeus committed:
Configure Gradle

gnahzeus committed:
Implement A-MoreOOP: Use More OOP

gnahzeus committed:
Implement A-Packages: Organize into Packages

gnahzeus committed:
Implement A-JUnit: Add JUnit Tests

gnahzeus committed:
Implement A-Jar: Create a JAR File

gnahzeus committed:
Implement A-CodingStandard: Follow the Coding Standard

gnahzeus committed:
Implement A-JavaDoc: Add JavaDoc comments

gnahzeus committed:
Implement Level 9. Find

gnahzeus committed:
Merge branch 'branch-A-JavaDoc'

gnahzeus committed:
Merge branch 'branch-A-CodingStandard'

gnahzeus committed:
Merge branch 'branch-Level-9'

gnahzeus committed:
Fix style problems

gnahzeus committed:
Implement Level 10. GUI

grumphrey commented on src/main/java/zeus/main/Ui.java:
```diff
@@ -0,0 +1,133 @@
+    /**
+     * Prints exit message for user.
+     */
+    public void exitMessage() {
+        printFormatted("Bye. Hope to see you again soon!");
+    }
```
```suggestion
    public void showExitMessage() {
        printFormatted("Bye. Hope to see you again soon!");
    }
```

Should this method be named as a verb instead, e.g. `showExitMessage()` ?

grumphrey commented on src/main/java/zeus/command/DeleteCommand.java:
```diff
@@ -0,0 +1,44 @@
+     *
+     * @param idx Index of Task to be deleted in task list
+     */
+
```
```suggestion
```
Might this blank line between the javadocs and method be unnecessary?

grumphrey commented on src/main/java/zeus/command/TaskDoneCommand.java:
```diff
@@ -0,0 +1,47 @@
+    /**
+     * Constructor for TaskDoneCommand.
+     *
+     * @param idx
```
```suggestion
     * @param idx The index of the task to be marked as done.
```

Should a javadocs description be included for the parameter?

grumphrey commented on src/main/java/zeus/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,63 @@
+    /**
+     * Constructor of AddDeadlineCommand class.
+     *
+     * @param description Description of Deadline task
+     * @param by String that represents due date
```
```suggestion
     * @param description Description of Deadline task.
     * @param by String that represents due date.
```
Might be better to add a punctuation behind each parameter description

grumphrey commented on src/main/java/zeus/main/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+    }
+
+    /**
+     * Sets the task at the given index as done.
+     * @param idx Index of target Task
```
```suggestion
     * Sets the task at the given index as done.
     
     * @param idx Index of target Task
```

Might be good to add an empty line between the javadocs' description and parameter section!

grumphrey reviewed (commented):
Hey @gnahzeus, this looks good to me! Besides a few nitpicks with the javadocs, I don't see much issues with your variable and method names and code quality. Good job!

johnrhimawan commented on src/main/java/zeus/command/AddEventCommand.java:
```diff
@@ -0,0 +1,63 @@
+
+
+/**
+ * Command that handles adding Deadline.
```
Perhaps this comment would be better suited using the object ```Event``` rather than ```Deadline```?

johnrhimawan commented on src/main/java/zeus/command/FindCommand.java:
```diff
@@ -0,0 +1,24 @@
+
+public class FindCommand extends Command {
+
+    private String s;
```
Would the variable ```s``` be more useful if a more descriptive name was given?

johnrhimawan commented on src/main/java/zeus/main/Parser.java:
```diff
@@ -0,0 +1,102 @@
+
+import java.time.LocalDate;
+
+import zeus.command.*;
```
Should the imported classes be listed explicitly instead?

johnrhimawan commented on src/main/java/zeus/main/Ui.java:
```diff
@@ -0,0 +1,133 @@
+    /**
+     * Prints exit message for user.
+     */
+    public void exitMessage() {
```
```suggestion
    public void printExitMessage() {
```

Perhaps the method could be named using a verb instead?

johnrhimawan commented on src/main/java/zeus/main/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+ */
+public class TaskList {
+
+    private ArrayList<Task> taskList;
```
```suggestion
    private ArrayList<Task> tasks;
```
Perhaps this could be in plural form since you are using a collection of objects.

johnrhimawan commented on src/main/java/zeus/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,63 @@
+package zeus.command;
```
I like how you organized your code into packages.

johnrhimawan commented on src/main/java/zeus/main/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+     *
+     * @return size of task list
+     */
+    public int size() {
```
```suggestion
    public int getSize() {
```
Would it be better for this method name to be of a verb?

johnrhimawan commented on src/main/java/zeus/main/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+     * @param s User search input
+     * @return ArrayList of matching tasks
+     */
+    public ArrayList<Task> find(String s) {
```
```suggestion
    public ArrayList<Task> findMatch(String s) {
```
Perhaps this method name could be of a more descriptive verb.

johnrhimawan commented on src/main/java/zeus/main/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+     * @return ArrayList of matching tasks
+     */
+    public ArrayList<Task> find(String s) {
+        ArrayList<Task> matching = new ArrayList<>();
```
```suggestion
        ArrayList<Task> matchingTasks = new ArrayList<>();
```
Would it be better to use plural form since you are using collections?

johnrhimawan reviewed (commented):
Overall, the code written is excellent with only a few violations of good code quality. It is worth noting that you have organized your code into packages well which improved readability. However, there were instances where the variable and method names could be named to better suit the coding standards. LGMT.

gnahzeus committed:
Edit branch-Level-10

gnahzeus committed:
Merge branch-Level-10

gnahzeus committed:
Use assert feature to document important assumptions

gnahzeus committed:
Make improvement to Code Quality

gnahzeus committed:
Merge pull request #1 from gnahzeus/branch-A-Assertions

Use assert feature to document important assumptions

gnahzeus committed:
Merge master branch into branch-A-CodeQuality

gnahzeus committed:
Merge pull request #2 from gnahzeus/branch-A-CodeQuality

Make improvements to Code Quality

gnahzeus committed:
Add Increment: A-CI

Use GitHub Actions to set up Continuous Integration (CI).

gnahzeus committed:
Improve code quality

Some implementations go against code quality guidelines.

Making tweaks causes the code more readable and clear.

Let's rename variables and method names to improve code quality.

gnahzeus committed:
Add undo feature

Cannot undo previous commands.

An undo command is more user-friendly as it allows users to quickly
undo their last command.

Let's provide a way to undo the last command.
Enable any number of undos, up until the first command is reached.

gnahzeus committed:
Merge branch 'undo-function'

gnahzeus committed:
Fix style issues

gnahzeus committed:
Improve adherence to code quality guidelines

gnahzeus committed:
Improve adherence to javadocs style

gnahzeus committed:
Fix style issues

gnahzeus committed:
Finalize the features

gnahzeus committed:
Add User Guide

gnahzeus committed:
Finalise changes

gnahzeus committed:
Add files via upload

gnahzeus committed:
Add title to application

gnahzeus committed:
Merge branch 'master' of https://github.com/gnahzeus/ip

gnahzeus committed:
Remove Ui.png

gnahzeus committed:
Delete Ui.png

gnahzeus committed:
Add files via upload

gnahzeus committed:
Delete Screenshot 2022-09-16 at 5.31.40 PM.png

gnahzeus committed:
Add files via upload

gnahzeus committed:
Delete Ui.png

gnahzeus committed:
Add files via upload

gnahzeus committed:
Fix javadocs details

gnahzeus committed:
Merge branch 'master' of https://github.com/gnahzeus/ip
